api/v1alpha1/types_etcd.go

```diff
@@ -485,3 +567,45 @@ func (e *Etcd) GetRoleName() string {
 func (e *Etcd) GetRoleBindingName() string {
 	return fmt.Sprintf("%s:etcd:%s", GroupVersion.Group, e.Name)
 }
+
+// IsBackupStoreEnabled returns true if backup store has been enabled for the Etcd resource, else returns false.
+func (e *Etcd) IsBackupStoreEnabled() bool {
+	return e.Spec.Backup.Store != nil
+}
+
+// IsMarkedForDeletion returns true if a deletion timestamp has been set and false otherwise.
+func (e *Etcd) IsMarkedForDeletion() bool {
+	return !e.DeletionTimestamp.IsZero()
+}
+
+// GetSuspendEtcdSpecReconcileAnnotationKey gets the annotation key set on an Etcd resource signalling the intent
+// to suspend spec reconciliation for this Etcd resource. If no annotation is set then it will return nil.
+func (e *Etcd) GetSuspendEtcdSpecReconcileAnnotationKey() *string {
+	if metav1.HasAnnotation(e.ObjectMeta, SuspendEtcdSpecReconcileAnnotation) {
+		return pointer.String(SuspendEtcdSpecReconcileAnnotation)
+	}
+	if metav1.HasAnnotation(e.ObjectMeta, IgnoreReconciliationAnnotation) {
+		return pointer.String(IgnoreReconciliationAnnotation)
+	}
+	return nil
+}
+
+// IsReconciliationSuspended returns true if the Etcd resource has the annotation set to suspend spec reconciliation,
```

Comment:
The docstring is correct, since this annotation is used only to suspend spec reconciliation, and not status reconciliation. I will rather rename the method to `IsSpecReconciliationSuspended`, which makes more sense. Thanks.

Reply:
Status reconciliation means reconciling the status, ie, updating the status with latest state information from the etcd cluster. So we distinctly distinguish between spec and status reconciliation in etcd controller.